### PR TITLE
Add dynamic margin support for texts and drawables in toolbar

### PR DESCRIPTION
**Pull request by gulerbu**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Supports dynamic margins for toolbar texts and icons

## How Has This Been Tested?
Tested on emulator by giving margin values

### CODE REVIEW

File: libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/Toolbar.kt [hunk 1]
```diff
@@ -50,6 +52,10 @@ class Toolbar : ConstraintLayout {
         binding.textMiddle.setOnClickListener { middleTextClickListener?.invoke() }
         binding.textRightUp.setOnClickListener { upperRightTextClickListener?.invoke() }
         binding.textRightDown.setOnClickListener { lowerRightTextClickListener?.invoke() }
+        leftTextDefaultMargin =
+            resources.getDimensionPixelOffset(R.dimen.trendyol_uicomponents_toolbar_margin_left_side_text)
+        rightImageDrawableDefaultEndMargin =
+            resources.getDimensionPixelOffset(R.dimen.trendyol_uicomponents_toolbar_margin_right_side_icon)
```
Comment (bilgehankalkan) [inline libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/Toolbar.kt:58]:
```suggestion
```

File: libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/ToolbarViewState.kt [hunk 2]
```diff
@@ -19,7 +19,14 @@ data class ToolbarViewState(
...
     @StyleRes val lowerRightTextAppearance: Int = R.style.Trendyol_UIComponents_Toolbar_Text_LowerAction,
-    @DrawableRes val toolbarBackground: Int = android.R.color.white
+    @DrawableRes val toolbarBackground: Int = android.R.color.white,
+    val upperLeftTextMarginStartInPixel: Int? = null,
+    val lowerLeftTextMarginStartInPixel: Int?= null,
+    val upperRightTextMarginEndInPixel: Int?= null,
+    val lowerRightTextMarginEndInPixel: Int?= null,
+    val rightImageDrawableMarginEndInPixel: Int?= null,
+    val leftImageDrawableMarginStartInPixel: Int?= null
+
```
Comment (bilgehankalkan) [inline libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/ToolbarViewState.kt:29]:
```suggestion
    val upperLeftTextMarginStartInPixel: Int = 0,
    val lowerLeftTextMarginStartInPixel: Int = 0,
    val upperRightTextMarginEndInPixel: Int = 0,
    val lowerRightTextMarginEndInPixel: Int = 0,
    val rightImageDrawableMarginEndInPixel: Int = 0,
    val leftImageDrawableMarginStartInPixel: Int = 0
```

File: libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/ToolbarViewState.kt [hunk 3]
```diff
@@ -19,7 +19,14 @@ data class ToolbarViewState(
     @StyleRes val middleTextAppearance: Int = R.style.Trendyol_UIComponents_Toolbar_Text_Title,
     @StyleRes val upperRightTextAppearance: Int = R.style.Trendyol_UIComponents_Toolbar_Text_UpperAction,
     @StyleRes val lowerRightTextAppearance: Int = R.style.Trendyol_UIComponents_Toolbar_Text_LowerAction,
-    @DrawableRes val toolbarBackground: Int = android.R.color.white
+    @DrawableRes val toolbarBackground: Int = android.R.color.white,
```
Comment (bsobe) [inline libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/ToolbarViewState.kt:22]: is this parameter used?

Comment (gulerbu) [reply libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/ToolbarViewState.kt:22]: It is used to set background image resource drawable in toolbar layout

File: libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/ToolbarViewState.kt (hunk 2 above)
Comment (gulerbu) [reply libraries/toolbar/src/main/java/com/trendyol/uicomponents/toolbar/ToolbarViewState.kt:29]: When these are set as null it prevents binding adapter to set 0 for margin values

**Pull request merged**